### ISSUE TITLE
`@aws-sdk/client-s3` issue with worker targets

### ISSUE THREAD

**Issue by fayazara** [unjs/nitro#2621]
### Environment

nitropack: latest
Running on Cloudflare pages

### Reproduction

https://github.com/fayazara/nitro-s3

### Describe the bug

Using AWS client S3 sdk does not work on Cloudflare Pages, using any method from the S3 sdk throws an error from the obfuscated method name either `t is not a function` (when actually deployed on cloudflare pages or `t2 is not a function` when using wrangler locally

I have tested the same version of the sdk using Hono on Cloudflare page which works fine.

### Additional context

There was a node compatibility polyfill issue which @pi0 solved yesterday in https://github.com/unjs/unenv/issues/279, this new error `t2 is not a function` started popping up after this issue was fixed

### Logs

```sh
Error uploading file: TypeError: t2 is not a function

      at Wd.slurpFile
  (file:///Users/projects/nitro-s3/.wrangler/tmp/pages-mkxmpG/chunks/routes/index.post.mjs:1:242780)
      at loadSharedConfigFiles
  (file:///Users/projects/nitro-s3/.wrangler/tmp/pages-mkxmpG/chunks/routes/index.post.mjs:1:245724)
      at n
  (file:///Users/projects/nitro-s3/.wrangler/tmp/pages-mkxmpG/chunks/routes/index.post.mjs:1:247712)
      at null.<anonymous>
  (file:///Users/projects/nitro-s3/.wrangler/tmp/pages-mkxmpG/chunks/routes/index.post.mjs:1:188559)
      at async coalesceProvider
  (file:///Users/projects/nitro-s3/.wrangler/tmp/pages-mkxmpG/chunks/routes/index.post.mjs:1:188781)
      at null.<anonymous> (async
  file:///Users/projects/nitro-s3/.wrangler/tmp/dev-1Ax2UV/t7mjr99ri8n.js:9569:101)
  Error: Failed to upload file to bucket-name/filename.png
      at FileStorage.upload
  (file:///Users/projects/nitro-s3/.wrangler/tmp/pages-mkxmpG/chunks/routes/index.post.mjs:1:760528)
      at async Object.handler
  (file:///Users/projects/nitro-s3/.wrangler/tmp/pages-mkxmpG/chunks/routes/index.post.mjs:1:761772)
      at async Object.handler
  (file:///Users/projects/nitro-s3/.wrangler/tmp/pages-mkxmpG/chunks/runtime.mjs:1:72110)
      at null.<anonymous> (async
  file:///Users/projects/nitro-s3/.wrangler/tmp/dev-1Ax2UV/t7mjr99ri8n.js:17154:7)
      at null.<anonymous> (async
  file:///Users/projects/nitro-s3/.wrangler/tmp/dev-1Ax2UV/t7mjr99ri8n.js:19166:24)
      at async jsonError
  (file:///Users/main-user/Library/Caches/pnpm/dlx/cbftox4creh25irebxwmcy7fxe/190bf21ed13-b177/node_modules/.pnpm/wrangler@3.65.0/node_modules/wrangler/templates/middleware/middleware-miniflare3-json-error.ts:22:10)
      at async drainBody
  (file:///Users/main-user/Library/Caches/pnpm/dlx/cbftox4creh25irebxwmcy7fxe/190bf21ed13-b177/node_modules/.pnpm/wrangler@3.65.0/node_modules/wrangler/templates/middleware/middleware-ensure-req-body-drained.ts:5:10)
```

**Comment by pi0** [comment]
Thanks for the reproduction (TIP: you can do `pnpm build --no-minify && pnpm wrangler pages dev dist/` to have better insights). 

Stack trace without minify:

```
index.mjs:27562 Error uploading file: TypeError: readFile is not a function
    at slurpFile (index.mjs:11758:34)
    at loadSharedConfigFiles (index.mjs:11914:27)
    at index.mjs:12027:83
    at index.mjs:8789:33
    at async coalesceProvider (index.mjs:8816:18)
    at async index.mjs:8827:20
```

---

TLDR: `@aws-sdk/client-s3` uses Node.js APIs and depends on `node:fs/promises` which is not supported in worker environments. Nitro/unenv tries a lightweight polyfill but it is not enough.

The proper solution for `@aws-sdk/client-s3` to be fixed is that they add an `exports` field to their `package.json` with `workerd` and other conditions pointing to an ESM native compatible code.

Below is a workaround but I do highly recommend considering alternative / web-compatible clients such as [aws4fetch](https://www.npmjs.com/package/aws4fetch) ([example](https://developers.cloudflare.com/r2/examples/aws/aws4fetch/)) to avoid such issues. In the future unstorage/storage layer of Nitro will also adopt this approach. 

```js
export default defineNitroConfig({
  srcDir: "server",
  "preset": "cloudflare-pages",
  minify: false,
  rollupConfig: {
    plugins: [
      {
        // Workaround for @aws-sdk/client-s3 compatibility with Cloudflare Pages
        // Consider using aws4fetch or another HTTP client for better compatibility
        name: 'fix-aws-sdk-import',
        resolveId: {
          order: 'pre',
          handler(source, importer) {
              if (source === './runtimeConfig' && importer?.includes('@aws-sdk')) {
                return this.resolve('./runtimeConfig.browser', importer);
              }
          },
        }
      }
    ]
  }
});
```

**Comment by fayazara** [comment]
Thanks for the workaround @pi0 🙌
